### PR TITLE
fix: projects invalidation and data resetting on 404

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/3226

### Description:
- invalidate projects on delete
- reset projects on 404
- icon updates
- fix bug with localstorage saved project UUID when project gets deleted

https://user-images.githubusercontent.com/962095/194526440-fe17c6ad-a7d1-4aba-9720-cc7701bcdbea.mp4

### CODE REVIEW

File: packages/frontend/src/components/UserSettings/ProjectManagementPanel/index.tsx [hunk 1]
```diff
@@ -70,7 +72,7 @@ const ProjectListItem: FC<{
                         })}
                     >
                         <Button
-                            icon="delete"
+                            icon="trash"
```
Comment (rephus) [inline packages/frontend/src/components/UserSettings/ProjectManagementPanel/index.tsx:75]: why changing the icon ? I think `delete` is our standard

File: packages/frontend/src/hooks/useProjects.ts [hunk 2]
```diff
@@ -42,17 +55,16 @@ export const deleteLastProject = () => {
...
+
+    const defaultProject = query.data?.find(
         ({ type }) => type === ProjectType.DEFAULT,
     );
+
     return {
-        ...projectsQuery,
-        data: defaultProject || projectsQuery.data?.[0],
+        ...query,
+        originalData: query.data,
```
Comment (rephus) [inline packages/frontend/src/hooks/useProjects.ts:66]: what's this for ?

File: packages/frontend/src/components/UserSettings/ProjectManagementPanel/index.tsx (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/src/components/UserSettings/ProjectManagementPanel/index.tsx:75]: x in a circle is ugly, I can change to X but trash is more obvious

File: packages/frontend/src/hooks/useProjects.ts (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/hooks/useProjects.ts]: no reason, removed

**Review by rephus — APPROVED** [review]

**Pull request merged**